storage/indexer.go

```diff
@@ -183,9 +183,13 @@ func (i *Indexer) updateIndex(ctx context.Context) error {
 		metrics.StorageIndexerUpdateIndexErrorsTotal.Inc()
 		return fmt.Errorf("can't load the search-index-all index content: %w", err)
 	}
+	if anIndex == nil {
+		i.logger.Info("Downloaded new search-index-all index. No packages found.")
+		return nil
+	}
```

Comment:
Not sure if this is totally necessary. 
It looks like that `loadSearchIndexAll` function will not return a nil value
https://github.com/elastic/package-registry/blob/288bf96f2b7e520d333db91693fbb156395b8939/storage/index.go#L51-L91

If it is needed, taking into account that `err` is nil here, should it be returned an error instead in this scenario with no packages?

Reply:
At least, returning early here will make sure that the package-registry service will not delete all the packages from its memory. That will ensure that the service could continue with the previous packages if it finds `anIndex == nil`
WDYT ?